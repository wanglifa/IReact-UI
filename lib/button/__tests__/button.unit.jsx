import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../button'
import { mount } from 'enzyme'
import Icon from "../../icon/icon";

describe('Button', () => {
  it('render successfully', () => {
    //渲染一个Button，因为Button是一个对象所以我们可以把它转成json
    const json = renderer.create(<Button type="info" size="mini">小改改</Button>).toJSON()
    //期待它去匹配Snapshot
    expect(json).toMatchSnapshot()
  })
  it('onClick', () => {
    const fn = jest.fn()
    const component = mount(<Button onClick={fn}>小改改</Button>)
    component.find('button').simulate('click')
    expect(fn).toBeCalled
  })
})