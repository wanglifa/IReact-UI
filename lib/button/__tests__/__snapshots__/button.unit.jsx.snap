// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render successfully 1`] = `
<button
  className="ireact-button ireact-button-info ireact-button-mini"
>
  小改改
</button>
`;
